### DISCUSSION TITLE
What exactly IS shadowmap?

### DISCUSSION THREAD

**Discussion by Trainmaster9977** [ScanMountGoat/Smush-Material-Research#35] [Q&A] [answered]
Hello! Taking a look at the annotated code and renderdoc and can't tell _exactly_ what shadowmap is, and how it differs from normal lighting. (like IN_shLighting). Like vaguely map of shadows and lighting but the exact details are unclear honestly.

**Comment by ScanMountGoat** [comment, marked as answer]
Shadow mapping records the closest depth of each point from the perspective of the light and the camera. The two depth values can be compared to determine if a pixel is "in shadow" or not. This requires a position attribute for both the regular position and a separate shadow map position. The "shadow map" is just the depth texture of the scene rendered from the light's perspective. This is a standard technique, which you can read about below.
https://learnopengl.com/Advanced-Lighting/Shadows/Shadow-Mapping

**Reply by Trainmaster9977** [reply]
ah, I think I understand. So it's not to do with how light casts onto an object, which is assumedly controlled by IN_shLighting.
